lib/temporalio/worker/workflow_worker.rb

```diff
@@ -0,0 +1,127 @@
+require 'temporal/sdk/core/workflow_commands/workflow_commands_pb'
+require 'temporalio/error/failure'
+require 'temporalio/errors'
+require 'temporalio/interceptor'
+require 'temporalio/interceptor/chain'
+require 'temporalio/worker/workflow_runner'
+
+module Temporalio
+  class Worker
+    # @api private
+    class WorkflowWorker
+      def initialize(namespace, task_queue, worker, workflows, converter, interceptors)
+        @namespace = namespace
+        @task_queue = task_queue
+        @worker = worker
+        @workflows = prepare_workflows(workflows)
+        @converter = converter
+        @inbound_interceptors = Temporalio::Interceptor::Chain.new(
+          Temporalio::Interceptor.filter(interceptors, :workflow_inbound),
```

Comment:
I think filtering interceptors should be done in caller.

Reply:
I can see arguments for either approach. Treating as nit, but please feel free to refactor if you feel strongly about it.